assets/js/base/components/block-error-boundary/block-error.tsx

```diff
@@ -2,9 +2,17 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import PropTypes from 'prop-types';
 import { WC_BLOCKS_IMAGE_URL } from '@woocommerce/block-settings';
 
+interface BlockErrorProps {
+	imageUrl: string;
+	header: string;
+	text: React.ReactNode;
+	errorMessage: string | null;
+	errorMessagePrefix: string;
+	button: React.ReactNode;
+}
+
```

Comment:
I would suggest copying the JSDoc associated with the properties, they are still useful! They appear in Intellisense and also will appear in the Storybook docs.

Also, reading the `propTypes` it appears that `header, imageUrl, text` can be: `string | null | undefined` from the docs. In your case they are required strings.